packages/rest/src/http-handler.ts

```diff
@@ -33,8 +34,12 @@ export class HttpHandler {
     this.handleRequest = (req, res) => this._handleRequest(req, res);
   }
 
-  registerController(name: ControllerClass, spec: ControllerSpec) {
-    this._routes.registerController(name, spec);
+  registerController(
+    controllerCtor: ControllerClass,
+    spec: ControllerSpec,
+    factory?: ControllerFactory,
+  ) {
+    this._routes.registerController(controllerCtor, spec);
```

Comment:
`factory` argument is ignored, I think you need to pass it to `this._routes.registerController`? Please add a test that's failing with the current implementation because of that bug, it will serve as a regression guard for the future.